### DISCUSSION TITLE
draw geojson / text over terrain

### DISCUSSION THREAD

**Discussion by farfromrefug** [visgl/deck.gl#5649] [Q&A] [answered]
I have this 
<img width="727" alt="Screenshot 2021-04-10 at 17 14 23" src="https://user-images.githubusercontent.com/655344/114274853-98e97600-9a20-11eb-8e15-b607adae86e5.png">
To draw map mountain peaks.
Now i would like to draw those points at their "elevation" so that i see them over my terrain layer
<img width="727" alt="Screenshot 2021-04-10 at 17 18 21" src="https://user-images.githubusercontent.com/655344/114274901-c6362400-9a20-11eb-8e89-760f5fb5f00b.png">
I was thinking getElevation would do it but it does not seem so.
Is there a way to specify a "feature" elevation for drawing ?

**Comment by farfromrefug** [comment, marked as answer]
I actually got it by uzing the z coordinate in `getPosition`
```
getPosition: d => [...d.geometry.coordinates, d.properties.ele || 0]
```
It works perfectly with `TextLayer`. However it does not seem to work with `GeoJsonLayer`. Is this expected?

**Reply by Pessimistress** [reply]
You can add altitude to positions by using 3D coordinates in the GeoJSON.
`getElevation` does not refer to the anchor position, and there is no `getPosition` prop in the GeoJsonLayer. See [documentation](https://deck.gl/docs/api-reference/layers/geojson-layer).

**Reply by farfromrefug** [reply]
@Pessimistress i see. would nt it be good yo add it ? i mean when you get geojson from mvt layer it would be much easier to have getPosition and then use it in the underlying layers i think

**Reply by Pessimistress** [reply]
The purpose of the GeoJsonLayer is to support the GeoJSON specification. If you don't want to encode your data in GeoJSON then supply your own [renderSubLayers](https://deck.gl/docs/api-reference/geo-layers/tile-layer#rendersublayers) and use any other layer.

**Reply by farfromrefug** [reply]
@Pessimistress i understand what you mean. And yes i can use renderSubLayers. I was just saying adding `getPosition` would not break anything (i think) and allow people to do 3d maps like i am trying to do.
But thanks and i got my answer
